pkgs/development/libraries/qt-6/6.2/default.nix

```diff
@@ -0,0 +1,213 @@
+, gst-plugins-base
+, gtk3
+, dconf
+, llvmPackages_5
+, darwin
+
+  # options
+, developerBuild ? false
+, decryptSslTraffic ? false
+, debug ? false
+}:
+
+let
+
+  qtCompatVersion = srcs.qtbase.version;
+
+  stdenvActual = if stdenv.cc.isClang then llvmPackages_5.stdenv else stdenv;
+
+  mirror = "https://download.qt.io";
+  srcs = import ./srcs.nix { inherit fetchurl mirror; } // {
+    # qtwebkit does not have an official release tarball on the qt mirror and is
+    # mostly maintained by the community.
+    /* not-yet available for qt6
+    qtwebkit = rec {
+      src = fetchFromGitHub {
+      # https://github.com/qt/qtwebkit
+      owner = "qt";
+      repo = "qtwebkit";
+      rev = "v${version}";
+      sha256 = "0x8rng96h19xirn7qkz3lydal6v4vn00bcl0s3brz36dfs0z8wpg";
+      };
+      version = "5.212.0-alpha4";
+    };
+    */
+  };
+
+  qtModule =
+    import ../qtModule.nix
+      {
+        inherit perl lib cmake writeText;
+        # Use a variant of mkDerivation that does not include wrapQtApplications
+        # to avoid cyclic dependencies between Qt modules.
+        mkDerivation =
+          import ../mkDerivation.nix
+            { inherit lib debug; wrapQtAppsHook = null; }
+            stdenvActual.mkDerivation;
+      }
+      { inherit self srcs; };
+
+  addPackages = self: with self;
+    let
+      callPackage = self.newScope { inherit qtCompatVersion qtModule srcs; };
+    in
+    {
+
+      inherit callPackage qtCompatVersion qtModule srcs;
+
+      mkDerivationWith =
+        import ../mkDerivation.nix
+          { inherit lib debug; inherit (self) wrapQtAppsHook; };
+
+      mkDerivation = mkDerivationWith stdenvActual.mkDerivation;
+
+      # NOTE large packages:
+      # 150min qtdeclarative
+      # 90min qtbase
+      # 30min qtwayland
+      # qtquick3d?
+
+      qtbase = callPackage ../modules/qtbase.nix {
+        inherit (srcs.qtbase) src version;
+        inherit bison cups harfbuzz libGL dconf gtk3 developerBuild decryptSslTraffic;
+        withGtk3 = true;
+        inherit (darwin.apple_sdk.frameworks) AGL AppKit ApplicationServices Carbon Cocoa CoreAudio CoreBluetooth
+          CoreLocation CoreServices DiskArbitration Foundation OpenGL MetalKit IOKit;
+        inherit (darwin) libobjc;
+      };
+
+      qt3d = callPackage ../modules/qt3d.nix { };
+      qt5compat = callPackage ../modules/qt5compat.nix { };
+      qtcharts = callPackage ../modules/qtcharts.nix { };
+      qtconnectivity = callPackage ../modules/qtconnectivity.nix { };
+      qtdeclarative = callPackage ../modules/qtdeclarative.nix { };
+      qtdoc = callPackage ../modules/qtdoc.nix { };
+      qtimageformats = callPackage ../modules/qtimageformats.nix { };
+      #qtlocation = callPackage ../modules/qtlocation.nix {};
+      qtmultimedia = callPackage ../modules/qtmultimedia.nix {
+        inherit gstreamer gst-plugins-base;
+      };
+
+      # experiment
+      qtbaseDistbuild = callPackage ../modules/qtbase.distbuild.nix {
```

Comment:
This is probably meant for reviewing purposes, but:

```
pkgs/development/libraries/qt-6/modules/qtbase.distbuild.nix': No such file or directory
```

Reply:
thanks. that was an experiment with distributed compiling